File: nodes/modulo/modulo.process.py
async def process(inputs, settings, config, nodeConfig):
  # Validate inputs are numbers
  dividend = inputs.get("dividend")
  divisor = inputs.get("divisor")
  
  try:
    num_dividend = float(dividend)
  except (ValueError, TypeError):
    raise ValueError("Dividend must be a number")
    
  try:
    num_divisor = float(divisor)
  except (ValueError, TypeError):
    raise ValueError("Divisor must be a number")
    
  if num_divisor == 0:
    raise ValueError("Division by zero")
    
  # JavaScript-like modulo implementation
  # In JavaScript, the sign of the result matches the sign of the dividend
  # regardless of the sign of the divisor
  
  # Calculate absolute values
  abs_dividend = abs(num_dividend)
  abs_divisor = abs(num_divisor)
  
  # Calculate modulo with absolute values
  abs_result = abs_dividend % abs_divisor
  
  # Apply the sign of the dividend to the result
  if num_dividend < 0 and abs_result != 0:
    result = -abs_result
  else:
    result = abs_result
  
  # Round to 10 decimal places to match JavaScript implementation
  result = round(result, 10)
  
  return {
    "result": result
  } 

File: sdks/nodejs/nodes/modulo/modulo.process.js
export default async ({inputs, settings, config}) => {

  // Validate inputs are numbers
  const dividend = inputs.dividend;
  const divisor = inputs.divisor;
  
  if (dividend === null || dividend === undefined || isNaN(Number(dividend))) {
    throw new Error("Dividend must be a number");
  }
  
  if (divisor === null || divisor === undefined || isNaN(Number(divisor))) {
    throw new Error("Divisor must be a number");
  }
  
  const numDividend = Number(dividend);
  const numDivisor = Number(divisor);
  
  if (numDivisor === 0) {
    throw new Error("Division by zero");
  }


  let result = numDividend % numDivisor;

  // try to handle floating point division
  result = Number(result.toFixed(10));
  
  return {
    result: result
  };
}; 